core/src/storage_stage.rs

```diff
@@ -477,11 +478,13 @@ impl StorageStage {
                             .entry(account_id)
                             .or_default()
                             .push(proof.clone());
+                        proofs_collected += 1;
```

Comment:
I we should do this based on the storage_keys instead of adding another var. We can probably reset storage_keys every turn and just use storage_key_index to see if any proofs were collected. It should be a noop in the chacha fn if there are no keys. wdyt?

Reply:
storage_keys is a fixed size right now. we need to rework the logic for that. I just want to fix it in the easiest way right now and then we can change that later.

Reply:
Sounds good. But local cluster failed? Shouldn't have been impacted by this.

Reply:
Yea... got a look at that.

Reply:
Ah. process_turn updates the turn blockhash, can't skip doing that.